blog/2019-09-ngrx-data-views/README.md

```diff
@@ -0,0 +1,419 @@
+---
+title: "Handling normalized entities with data views"
+author: Danilo Hoffmann
+mail: dhhyi@aol.com
+published: 2019-09-30
+last-change: 2019-09-30
+keywords:
+  - Angular
+  - RxJS
+  - Reactive Programming
+  - Observables
+  - NgRx
+  - NgRx Entities
+  - Memoization
+language: en
+thumbnail: data-views-header.jpg
+hidden: false
+---
+
+Dealing with data supplied by enterprise systems can be quite tedious. Most of the time, data for a single entity has to be retrieved with multiple calls, because it is handled in a normalized way on the server. This Normalization however does a good job minimizing the space used up in any database by preventing duplication and with that also inconsistency. Have a look at the [Wikipedia Article on Database Normalization](https://en.wikipedia.org/wiki/Database_normalization) for more information.
+
+If the API giving us access to the data is just a simple REST API exposing all entities via separate endpoints, we have to accumulate the needed data in the front-end ourselves.
+
```

Comment:
This post already assumes that the store is a normalized database. This isn't necessarily the case. using it normalized is very cool, but I think there are scenarios where it's better to use the store as a simple data sink with redundancies.

Therefore, this article could/should start with a short part about this. Outline:

## The NgRx store a a normalized database

- what is normalization
- backend also does n11n
- why not also do this in the frontend?
- see the store as a database
- frontend DB is a "mirror" of the backend DB
- however: not always necessary
- sometimes store is just a pile of data


- for enterprise: more suitable to manage entities in the frontend as well
- bridge to next part

Reply:
I thought about this over and over again and I think the approach of the article is exactly that: enterprise software with an API that basically has a "normalized" API.
Later on, in the discussion about where to actually break this normalization is more suited to talk about this, I believe. I only want to mention different approaches on having redundancies in the store for further potential articles. :wink: